### PR TITLE
mwaa v5 migration

### PR DESCRIPTION
**Pull request by bbgu1**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Update mwaa blueprint to v5

### Motivation

EKS blueprint v5 changes

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: schedulers/terraform/managed-airflow-mwaa/eks.tf [hunk 1]
```diff
@@ -1,61 +1,115 @@
-#---------------------------------------------------------------
-# EKS Blueprints
-#---------------------------------------------------------------
-module "eks_blueprints" {
-  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.15.0"
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 19.13"
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/eks.tf:3]: replace the version with ` version = "~> 19.15"`

File: schedulers/terraform/managed-airflow-mwaa/eks.tf [hunk 2]
```diff
@@ -1,61 +1,115 @@
...
 
   cluster_name    = local.name
   cluster_version = var.eks_cluster_version
 
-  vpc_id             = module.vpc.vpc_id
-  private_subnet_ids = module.vpc.private_subnets
+  cluster_endpoint_private_access = true # if true, Kubernetes API requests within your cluster's VPC (such as node to control plane communication) use the private VPC endpoint
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/eks.tf:8]: remove this line as its a defult

File: schedulers/terraform/managed-airflow-mwaa/eks.tf [hunk 3]
```diff
@@ -1,61 +1,115 @@
...
+    # Recommended outbound traffic for Node groups
+    egress_all = {
+      description      = "Node all egress"
+      protocol         = "-1"
+      from_port        = 0
+      to_port          = 0
+      type             = "egress"
+      cidr_blocks      = ["0.0.0.0/0"]
+      ipv6_cidr_blocks = ["::/0"]
+    }
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/eks.tf:73]: remove this block as its default now

File: schedulers/terraform/managed-airflow-mwaa/eks.tf [hunk 4]
```diff
@@ -1,61 +1,115 @@
...
-      max_size               = 9
-      min_size               = 3
-      desired_size           = 3
-      create_launch_template = true
-      launch_template_os     = "amazonlinux2eks"
+  eks_managed_node_groups = {
+    #  We recommend to have a MNG to place your critical workloads and add-ons
+    #  Then rely on Karpenter to scale your workloads
+    #  You can also make uses on nodeSelector and Taints/tolerations to spread workloads on MNG or Karpenter provisioners
+    mng1 = {
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/eks.tf:80]: replace `mng1` with `core_node_group`

File: schedulers/terraform/managed-airflow-mwaa/eks.tf [hunk 5]
```diff
@@ -64,84 +118,22 @@ module "eks_blueprints" {
...
-    type = "EKS"
-
-    info {
-      eks_info {
-        namespace = "emr-mwaa"
-      }
-    }
-  }
-}
 #------------------------------------------------------------------------
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/eks.tf:127]: move this code to new file `mwaa.tf`

File: schedulers/terraform/managed-airflow-mwaa/main.tf [hunk 6]
```diff
@@ -4,7 +4,9 @@
 
 module "mwaa" {
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/main.tf:5]: move all mwaa resources to `mwaa.tf` file. 

Delete providers file and move provider code, locals and data sources to `main.tf`

File: schedulers/terraform/managed-airflow-mwaa/providers.tf [hunk 7]
```diff
@@ -3,15 +3,15 @@ provider "aws" {
...
 
 provider "helm" {
   kubernetes {
-    host                   = module.eks_blueprints.eks_cluster_endpoint
-    cluster_ca_certificate = base64decode(module.eks_blueprints.eks_cluster_certificate_authority_data)
+    host                   = module.eks.cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
     token                  = data.aws_eks_cluster_auth.this.token
   }
 }
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/providers.tf:17]: Move these providers to top of the main.tf and remove this file

File: schedulers/terraform/managed-airflow-mwaa/addons.tf [hunk 8]
```diff
@@ -0,0 +1,87 @@
+#------------------------------------------------------------------------
+# EKS Addons
+#------------------------------------------------------------------------
+module "eks_blueprints_kubernetes_addons" {
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/addons.tf:4]: replace this name from `eks_blueprints_kubernetes_addons ` to `eks_blueprints_addons `

File: schedulers/terraform/managed-airflow-mwaa/data.tf [hunk 9]
```diff
@@ -1,5 +1,5 @@
 data "aws_eks_cluster_auth" "this" {
-  name = module.eks_blueprints.eks_cluster_id
+  name = module.eks.cluster_name
 }
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/data.tf:4]: This policy data "aws_iam_policy_document" "emr_on_eks"  might not be required with the new emr_containers module

File: schedulers/terraform/managed-airflow-mwaa/addons.tf [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  name        = format("%s-%s", local.name, "emr-job-iam-policies")
+  description = "IAM policy for EMR on EKS Job execution"
+  path        = "/"
+  policy      = data.aws_iam_policy_document.emr_on_eks.json
+}
+
+#----------------------------------------------------------------------------
+# EMR on EKS
+#----------------------------------------------------------------------------
+module "emr_containers" {
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/addons.tf:69]: move this module to new file called `emr-eks.tf`

File: schedulers/terraform/managed-airflow-mwaa/cleanup.sh [hunk 11]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+
+targets=(
+  "module.emr_containers"
+  "module.eks_blueprints_kubernetes_addons"
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/cleanup.sh:7]: check targets once all the updates are done

**Review by vara-bonthu — COMMENTED** [review]
Thanks @bbgu1 for the MWAA upgrade PR. 
I have left some comments to update.

**Comment by vara-bonthu** [comment]
@bbgu1 Could you please resolve conflicts and double check with our internal blueprint standards. I will review and merge it accordingly.

File: schedulers/terraform/managed-airflow-mwaa/cleanup.sh [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+# Terraform destroy full
+#-------------------------------------------
+terraform destroy -auto-approve
+destroy_output=$(terraform destroy -auto-approve 2>&1)
+if [[ $? -eq 0 && $destroy_output == *"Destroy complete!"* ]]; then
+  echo "SUCCESS: Terraform destroy of all targets completed successfully"
+else
+  echo "FAILED: Terraform destroy of all targets failed"
+  exit 1
+fi
```
[the remark this answers is not in the document]
Comment (bbgu1) [reply schedulers/terraform/managed-airflow-mwaa/cleanup.sh:53]: compared the files and they seemed identical

**Comment by bbgu1** [comment]
@vara-bonthu  all resolved.   Had quite a lot trouble this week due to the terraform-aws-modules/security-group/aws module failing to create rules for vpc_endpoints_sg during my tests.   Took a while to find the issue and I am still working on finding the cause of it.   For this blueprint I switched to use the terraform aws_security_group module which IMO is much cleaner.

File: schedulers/terraform/managed-airflow-mwaa/install.sh [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+echo "Initializing ..."
+terraform init || echo "\"terraform init\" failed"
+
+# List of Terraform modules to apply in sequence
+targets=(
+  "module.vpc"
+  "module.mwaa"
+  "module.eks"
+  "module.eks_blueprints_addons"
+  "module.datahub"
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/install.sh:15]: Do we need this module?

File: schedulers/terraform/managed-airflow-mwaa/mwaa.tf [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+}
+
+resource "aws_iam_role_policy_attachment" "this" {
+  role       = module.mwaa.mwaa_role_name
+  policy_arn = aws_iam_policy.this.arn
+}
+
+#------------------------------------------------------------------------
+# Security group to open access to mwaa
+#------------------------------------------------------------------------
```
Comment (vara-bonthu) [inline schedulers/terraform/managed-airflow-mwaa/mwaa.tf:84]: we can remove this if its not required

**Review by vara-bonthu — COMMENTED** [review]
@bbgu1 LGTM 👍🏼  Two minor comments and thenwe are good to merge.

**Comment by vara-bonthu** [comment]
> @vara-bonthu all resolved. Had quite a lot trouble this week due to the terraform-aws-modules/security-group/aws module failing to create rules for vpc_endpoints_sg during my tests. Took a while to find the issue and I am still working on finding the cause of it. For this blueprint I switched to use the terraform aws_security_group module which IMO is much cleaner.

Thank you for upgrading this blueprint and fixing the issue 👍🏼

**Comment by bbgu1** [comment]
Fixed the two files you mentioned - thanks for reviewing.

**Review by vara-bonthu — APPROVED** [review]
LGTM👍

**Pull request merged**